src/videojs-contrib-hls.js

```diff
@@ -117,110 +117,107 @@ const handleHlsLoadedMetadata = function(qualityLevels, hls) {
 };
 
 /**
+ * Resuable stable sort function
+ *
+ * @param {Playlists} playlists
+ * @param {Function} different comparators
+ * @function stableSort
+ */
+const stableSort = function(array, sortFn) {
+  let cmp;
+
+  array.sort(function(left, right) {
+    cmp = sortFn(left, right);
+
+    if (cmp === 0) {
+      return array.indexOf(left) - array.indexOf(right);
```

Comment:
`Array.sort` works on the array in place, so the indices of items will change as this function is sorting. You need to use the original indices of the items in the array for stability. Since we are trying to avoid adding properties to the original elements, I'd suggest creating a copy of the array before sorting to check `indexOf` on